### ISSUE TITLE
When specify java-cataloger, java-pom-cataloger will also be selected

### ISSUE THREAD

**Issue by evanchaoli** [anchore/syft#2136]
**What happened**:

I need to scan a directory, but I want to use only image catalogers, so I tried to use `--catalogers` options, like:

```
$syft <path-to-scan> --catalogers=...,java-cataloger,graalvm-native-image-cataloger,...
```

As you can see, I only specified "java-cataloger", but scan result always contained packaged found by "java-pom-cataloger". So I debugged a little bit, and found the problem is at https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/cataloger.go#L186C56-L186C56. According to the function, "java-pom-cataloger" will match to "java-cataloger", so that it is selected as well.

I wonder if the behavior is intentional? If yes, why? @wagoodman 

**What you expected to happen**:

**Steps to reproduce the issue**:

**Anything else we need to know?**:

**Environment**:
- Output of `syft version`: main branch
- OS (e.g: `cat /etc/os-release` or similar): MacOS

**Comment by tgerla** [comment]
Hi @evanchaoli, thanks for the report, we've added this to a list of related issues to be fixed as soon as we can.

**Comment by wagoodman** [comment]
This is not intentional behavior and will be addressed by #1383
